aiida/storage/sqlite_dos/backend.py

```diff
@@ -97,7 +98,7 @@ class Configuration(BaseModel):
         filepath: str = Field(
             title='Directory of the backend',
             description='Filepath of the directory in which to store data for this backend.',
-            default_factory=mkdtemp
+            default_factory=lambda: AIIDA_CONFIG_FOLDER / 'repository' / f'sqlite_dos_{uuid4().hex}'
         )
```

Comment:
Was thinking of opening an issue, but perhaps we can just add this in this PR. Currently, the `filepath` is not resolved to the absolute path, so if the user provides a custom path during creation of the profile:

```
❯ verdi profile setup core.sqlite_dos -n --profile tmp --email no@email.com --filepath ../../tmp_storage
Report: Initialising the storage backend.
Report: Storage initialisation completed.

Success: Created new profile `tmp`.
```

This breaks when the user changes directory:

```
 ✘ storage:     Unable to instatiate profile's storage.
Error: FileNotFoundError: [Errno 2] No such file or directory: '../../tmp_storage/database.sqlite'
```

Maybe we can fix by adding a validator to the `filepath` field, e.g.:

```python
        @field_validator('filepath')
        @classmethod
        def ensure_absolute(cls, value):
            """Ensure that the filepath is absolute."""
            if not Path(value).is_absolute():
                return Path(value).resolve().as_posix()
```

Reply:
Indeed, thanks for testing, I already have these changes locally, together with some other fixes. Was about to make the PR